sklearn/utils/tests/test_utils.py

```diff
@@ -464,6 +466,21 @@ def test_safe_indexing_container_axis_0_unsupported_type():
         _safe_indexing(array, indices, axis=0)
 
 
+def test_safe_indexing_pandas_no_settingwithcopy_warning():
+    # Using safe_indexing with an array-like indexer gives a copy of the
+    # DataFrame -> ensure it doesn't raise a warning if modified
+    pd = pytest.importorskip("pandas")
+    if parse_version(pd.__version__) < parse_version("0.25.0"):
+        raise SkipTest(
+            "Older pandas version still raise a SettingWithCopyWarning warning"
+        )
+    X = pd.DataFrame({"a": [1, 2, 3], "b": [3, 4, 5]})
+    subset = _safe_indexing(X, [0, 1], axis=0)
+    with pytest.warns(None) as record:
+        subset.iloc[0, 0] = 10
+    assert len(record) == 0, f"{[str(rec.message) for rec in record]}"
+
```

Comment:
Maybe we could even add an assertion to check that the original dataframe `X` has not be mutated by line 480 and that thefore `_safe_indexing` with integer keys will have the same copy semantics on pands dataframes and numpy arrays.


```suggestion
    # The original dataframe is unaffected by the assignment on the subset:
    assert X.iloc[0, 0] = 1
```

Reply:
Yep, that can't hurt to check